lib/Dialect/HW/HWTypes.cpp

```diff
@@ -270,6 +276,17 @@ Type StructType::parse(AsmParser &p) {
   return get(p.getContext(), parameters);
 }
 
+LogicalResult StructType::verify(function_ref<InFlightDiagnostic()> emitError,
+                                 ArrayRef<StructType::FieldInfo> elements) {
+  llvm::SmallDenseSet<StringAttr> fieldNameSet;
+  fieldNameSet.reserve(elements.size());
+  for (const auto &elt : elements)
+    if (!fieldNameSet.insert(elt.name).second)
+      return emitError() << "duplicate field name '" << elt.name.getValue()
+                         << "' in hw.struct type";
+  return success();
```

Comment:
What about printing all errors and then erroring if there was any? This would provide an improved debugging experience. Ditto for `UnionType::verify`.